Build failure: netcat-openbsd

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

This is another casualty of the gcc 15 upgrade #475479.

### Can Hydra reproduce this build failure?

Yes

### Link to Hydra build job

https://hydra.nixos.org/build/318063950

### Relevant log output

```console
error: Cannot build '/nix/store/vn7zifhc12lvifxixwyvxixg65syfwi0-netcat-openbsd-1.219-1.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/hrd5rqxhd0nnd02ivf2cm6rwjxvvf6ki-netcat-openbsd-1.219-1
       Last 25 log lines:
       > patching file nc.1
       > patching file netcat.c
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/lw117lsr8d585xs63kx5k233impyrq7q-bash-5.3p3/bin/bash
       > gcc -g -O2 -c netcat.c -o netcat.o
       > netcat.c: In function 'readwrite':
       > netcat.c:1528:41: error: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
       >  1528 |                         signal(SIGALRM, quit);
       >       |                                         ^~~~
       >       |                                         |
       >       |                                         void (*)(void)
       > In file included from netcat.c:100:
       > /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:88:57: note: expected '__sighandler_t' {aka 'void (*)(int)'} but argument is of type 'void (*)(void)'
       >    88 | extern __sighandler_t signal (int __sig, __sighandler_t __handler)
       >       |                                          ~~~~~~~~~~~~~~~^~~~~~~~~
       > netcat.c:239:13: note: 'quit' declared here
       >   239 | static void quit();
       >       |             ^~~~
       > /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:72:16: note: '__sighandler_t' declared here
       >    72 | typedef void (*__sighandler_t) (int);
       >       |                ^~~~~~~~~~~~~~
       > make: *** [Makefile:17: netcat.o] Error 1
```

### Additional context

_No response_

### System metadata

```
# nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`
```

### Notify maintainers

@Artturin 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ne/netcat-openbsd/package.nix
+++ b/pkgs/by-name/ne/netcat-openbsd/package.nix
@@ -7,7 +7,7 @@
   installShellFiles,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "netcat-openbsd";
   version = "1.234-1";
 
@@ -15,7 +15,7 @@ stdenv.mkDerivation rec {
     domain = "salsa.debian.org";
     owner = "debian";
     repo = "netcat-openbsd";
-    tag = "debian/${version}";
+    tag = "debian/${finalAttrs.version}";
     hash = "sha256-6pCsBbS2IjXyXgNXURHK3uMRTJ0aXAsu29kc7f479Os=";
   };
 
@@ -35,8 +35,7 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin
-    mv nc $out/bin/nc
+    install -Dm755 -t $out/bin nc
     installManPage nc.1
 
     runHook postInstall
@@ -57,4 +56,4 @@ stdenv.mkDerivation rec {
     # never built on aarch64-darwin, x86_64-darwin since first introduction in nixpkgs
     broken = stdenv.hostPlatform.isDarwin;
   };
-}
+})

--- a/pkgs/by-name/ne/netcat-openbsd/package.nix
+++ b/pkgs/by-name/ne/netcat-openbsd/package.nix
@@ -9,14 +9,14 @@
 
 stdenv.mkDerivation rec {
   pname = "netcat-openbsd";
-  version = "1.219-1";
+  version = "1.234-1";
 
   src = fetchFromGitLab {
     domain = "salsa.debian.org";
     owner = "debian";
     repo = "netcat-openbsd";
-    rev = "refs/tags/debian/${version}";
-    sha256 = "sha256-rN8pl3Qf0T8bXGtVH22tBpGY/EcnbgGm1G8Z2patGbo=";
+    tag = "debian/${version}";
+    hash = "sha256-6pCsBbS2IjXyXgNXURHK3uMRTJ0aXAsu29kc7f479Os=";
   };
 
   strictDeps = true;

